pkg/webhooks/machine_webhook.go

```diff
@@ -1332,6 +1335,58 @@ func validateMachineLifecycleHooks(m, oldM *machinev1.Machine) []error {
 	return errs
 }
 
+func validateAzureDataDisks(machineName string, spec *machinev1.AzureMachineProviderSpec, parentPath *field.Path) []error {
+
+	var errs []error
+	dataDiskLuns := make(map[int32]struct{})
+	dataDiskNames := make(map[string]struct{})
```

Comment:
Good use of empty struct for an efficient set!

Reply:
🙂